step-api/src/main/java/org/jenkinsci/plugins/workflow/structs/DescribableHelper.java

```diff
@@ -331,7 +342,7 @@ static ParameterType of(Type type) {
                     }
                 }
                 if (acceptsList(type)) {
-                    return new ArrayType(of(((ParameterizedType) type).getActualTypeArguments()[0]));
+                    return new ArrayType(List.class, of(((ParameterizedType) type).getActualTypeArguments()[0]));
```

Comment:
Might actually be `Collection`.

(Do you want to give `Type` rather than `Class`?)